### PR TITLE
update python samples

### PR DESCRIPTION
**Pull request by henrychi2**
1) Add hash-partitioned parallel sample (ported from Java sample)
2) Update grep sample to skip subdirectory entries when searching for files to read in a directory
3) Change logic for computing arithmetic mean and standard deviation in outlier sample to avoid keeping all tuples in memory

### CODE REVIEW

**Comment by ddebrunner** [comment]
Can one of the admins verify this patch?

File: samples/python/topology/state/find_outliers_functions.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import random
-import statistics
+import math
```
Comment (ddebrunner) [inline samples/python/topology/state/find_outliers_functions.py]: Why the change in packages?

Comment (henrychi2) [inline samples/python/topology/state/find_outliers_functions.py]: The `statistics` package has functions to compute the mean and stddev given a list of tuples.  Since we don't want to keep a history of all tuples in memory, we are changing the logic to compute the mean and stddev incrementally.  This requires some functions in the `math` library (like math.sqrt()), and the `statistics` package is no longer needed.
Computing the mean and stddev incrementally is already done by the Java sample.  I didn't do a proper port with the first delivery of the Python sample.
